nilearn/plotting/html_surface.py

```diff
@@ -20,26 +20,85 @@ class SurfaceView(HTMLDocument):
     pass
 
 
+def _mix_colormaps(fg, bg):
+    """Mixes foreground and background arrays of RGBA colors.
+
+    Args:
+        fg (numpy.ndarray): Array of shape (n, 4), foreground colors
+        bg (numpy.ndarray): Array of shape (n, 4), background colors
+
+    Returns:
+        numpy.ndarray: Array of shape (n, 4), mixed colors
+    """
+    # Adapted from https://stackoverflow.com/questions/726549/algorithm-for-additive-color-mixing-for-rgb-values/727339#727339 # noqa: E501
+    if fg.shape != bg.shape:
+        raise Exception(
+            "Trying to mix colormaps with different shapes: "
+            f"{fg.shape}, {bg.shape}"
+        )
+
+    mix = np.empty_like(fg)
+
+    mix[:, 3] = 1 - (1 - fg[:, 3]) * (1 - bg[:, 3])
+    mix[:, 0] = (
+        fg[:, 0] * fg[:, 3] / mix[:, 3]
+        + bg[:, 0] * bg[:, 3] * (1 - fg[:, 3]) / mix[:, 3]
+    )
+    mix[:, 1] = (
+        fg[:, 1] * fg[:, 3] / mix[:, 3]
+        + bg[:, 1] * bg[:, 3] * (1 - fg[:, 3]) / mix[:, 3]
+    )
+    mix[:, 2] = (
+        fg[:, 2] * fg[:, 3] / mix[:, 3]
+        + bg[:, 2] * bg[:, 3] * (1 - fg[:, 3]) / mix[:, 3]
+    )
+
+    return mix
+
+
 def _get_vertexcolor(surf_map, cmap, norm,
-                     absolute_threshold=None, bg_map=None):
-    vertexcolor = cmap(norm(surf_map).data)
-    if absolute_threshold is None:
-        return to_color_strings(vertexcolor)
+                     absolute_threshold=None, bg_map=None,
+                     bg_on_data=None, bg_map_rescale=None, darkness=None):
     if bg_map is None:
         bg_map = np.ones(len(surf_map)) * .5
         bg_vmin, bg_vmax = 0, 1
     else:
         bg_map = surface.load_surf_data(bg_map)
+
+    # scale background map if need be
+    if bg_map_rescale:
         bg_vmin, bg_vmax = np.min(bg_map), np.max(bg_map)
-    bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
-    bg_color = plt.get_cmap('Greys')(bg_norm(bg_map))
-    vertexcolor[np.abs(surf_map) < absolute_threshold] = bg_color[
-        np.abs(surf_map) < absolute_threshold]
-    return to_color_strings(vertexcolor)
+        bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
+        bg_data = bg_norm(bg_map)
+    else:
+        bg_data = bg_map
+
+    if darkness is not None:
+        bg_data *= darkness
+    bg_colors = plt.get_cmap('Greys')(bg_data)
+
+    # select vertices which are filtered out by the threshold
+    if absolute_threshold is None:
+        under_threshold = np.zeros_like(surf_map, dtype=bool)
+    else:
+        under_threshold = np.abs(surf_map) < absolute_threshold
+
+    surf_colors = cmap(norm(surf_map).data)
+    # set transparency of voxels under threshold to 0
+    surf_colors[under_threshold, 3] = 0
+    if bg_on_data:
+        # if need be, set transparency of voxels above threshold to 0.7
+        # so that background map becomes visible
+        surf_colors[~under_threshold, 3] = 0.7
```

Comment:
Should this become a parameter ?

Reply:
Eventually, this should become a parameter in another PR. As discussed above with Jérôme (I know it's a long discussion haha), we intend to deprecate `alpha`'s current behaviour (which controls the background image transparency) so that it can parametrize what you want here.